HW_9/Task 2.js

```diff
@@ -0,0 +1,66 @@
+// const characters = [
+//     { 'name': 'Barney', 'age': 36 },
+//     { 'name': 'Fred', 'age': 40 },
+//     { 'name': 'Jack', 'age': 50 }
+//   ];
+
+// 1. Напишите функцию addCharacter(character) позволяющую добавить новый объект в массив characters. 
+//     Объект должен иметь поля name (string) и age (number)
+// 2. Напишите функцию getCharacter(name), позволяющую получить объект персонажа по его имени// getCharacter('Fred') => { 'name': 'Fred', 'age': 40 }
+// 3. Напишите функцию getCharactersByAge(minAge), возвращающую массив персонажей НЕ МЛАДШЕ minAge // getCharactersByAge(40) => [{ 'name': 'Fred', 'age': 40 },{ 'name': 'Jack', 'age': 50 }]
+// 4. Напишите функцию updateCharacter(name, newCharacter). (Методом getCharacter(name) получаем ссыклку на нужного персонажа, а потом меняем ему данные)
+// 5. Напишите функцию для удаления персонажа removeCharacter(name) (Реализовать через splice, индекс персонажа искать методом findInxex)
+
+const characters = [
+    { 'name': 'Barney', 'age': 36 },
+    { 'name': 'Fred', 'age': 40 },
+    { 'name': 'Jack', 'age': 50 }
+];
+// 1. Напишите функцию addCharacter(character) позволяющую добавить новый объект в массив characters. 
+//     Объект должен иметь поля name (string) и age (number)
+function addCharacter(character) {
+    if (typeof character['name'] === 'string' && typeof character['age'] === 'number') {
+        characters.push(character);
+    } else {
+        console.log('Invalid fields');
+    }
+}
+addCharacter({ 'name': 'Phill', 'age': 45 });
+console.log(characters);
+
+// 2. Напишите функцию getCharacter(name), позволяющую получить объект персонажа по его имени// getCharacter('Fred') => { 'name': 'Fred', 'age': 40 }
+
+function getCharacter(charName = '') {
+    if (characters.map(element => Object.values(element)).flat().includes(charName)) {
+        return characters.find(character => character.name === charName);
+    } else console.log('Not found');
```

Comment:
Не совсем понятно для чего нам в условии нужем массив, ведь `map` возвращает именно его, однако все же проще сразу искать и если не нашли, соответствующую ошибку 
```suggestion
    const character = characters.find(character => character.name === charName);
    if (!character) {
       console.log('Not found');
     } else {
         return character;
      }
```